File: servers/google-workspace/server-mock.py
#!/usr/bin/env python3
"""Mock Google Workspace MCP for local testing"""

from fastapi import FastAPI
import os

app = FastAPI(title="Google Workspace MCP (Test)")

@app.get("/health")
async def health():
    return {"status": "ok", "service": "google-workspace-mcp-test"}

@app.get("/")
async def root():
    return {
        "service": "Google Workspace MCP",
        "version": "1.0.0-test",
        "tools_available": [
            "docs", "sheets", "slides", "drive", "gmail", "calendar"
        ],
        "note": "Mock server - full version requires OAuth setup"
    }

@app.post("/docs")
async def list_docs():
    return {
        "success": False,
        "message": "Requires GOOGLE_OAUTH_CLIENT_ID and GOOGLE_OAUTH_CLIENT_SECRET",
        "error": "Not configured for mock testing"
    }

if __name__ == "__main__":
    import uvicorn
    port = int(os.getenv("MCP_HTTP_PORT", "8000"))
    host = os.getenv("MCP_HOST", "0.0.0.0")
    print(f"Starting Google Workspace MCP (mock) on {host}:{port}")
    uvicorn.run(app, host=host, port=port)

File: servers/dokploy/server-mock.js
#!/usr/bin/env node
/**
 * Mock Dokploy MCP for local testing
 */

const http = require('http');

const port = parseInt(process.env.MCP_HTTP_PORT || '8000');
const host = process.env.MCP_HOST || '0.0.0.0';

const server = http.createServer((req, res) => {
  res.setHeader('Content-Type', 'application/json');

  if (req.url === '/health' && req.method === 'GET') {
    res.writeHead(200);
    res.end(JSON.stringify({
      status: 'ok',
      service: 'dokploy-mcp-test'
    }));
  } else if (req.url === '/' && req.method === 'GET') {
    res.writeHead(200);
    res.end(JSON.stringify({
      service: 'Dokploy MCP',
      version: '1.0.0-test',
      tools_available: [
        'list-projects', 'list-services', 'deploy', 'logs', 'status'
      ],
      note: 'Mock server - full version requires DOKPLOY_URL and DOKPLOY_API_KEY'
    }));
  } else {
    res.writeHead(404);
    res.end(JSON.stringify({
      success: false,
      message: 'Requires DOKPLOY_URL and DOKPLOY_API_KEY',
      error: 'Not configured for mock testing'
    }));
  }
});

server.listen(port, host, () => {
  console.log(`Starting Dokploy MCP (mock) on ${host}:${port}`);
});
